src/main/java/graphql/schema/SchemaTransformer.java

```diff
@@ -151,9 +151,13 @@ private Object transformImpl(final GraphQLSchema schema, GraphQLSchemaElement sc
         }
 
         if (schema != null) {
-            GraphQLSchema graphQLSchema = dummyRoot.rebuildSchema(codeRegistry);
-            if (postTransformation != null) {
-                graphQLSchema = graphQLSchema.transform(postTransformation);
+
+            GraphQLSchema graphQLSchema = schema;
+            if (schemaChanged || codeRegistry.hasChanged()) {
+                graphQLSchema = dummyRoot.rebuildSchema(codeRegistry);
+                if (postTransformation != null) {
+                    graphQLSchema = graphQLSchema.transform(postTransformation);
+                }
```

Comment:
This is a BIG change - if the visitor didnt change anything and the code registry didn't change - then nothing has changed - its the same schema - so we return it direct